File: backend/local-api-module/src/main/java/org/vyatsu/localApiModule/dto/response/api/UserDto.java
package org.vyatsu.localApiModule.dto.response.api;

import lombok.*;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * DTO for {@link org.vyatsu.localApiModule.entity.user.User}
 */

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserDto implements Serializable {
    Long id;
    String email;
    String password;
    String firstName;
    String secondName;
    String phoneNum;
    Boolean isActive;
    LocalDateTime createdAt;
}

File: app/src/main/java/com/bytebusters/android/mpsystem/dto/response/api/UserDto.kt
package com.bytebusters.android.mpsystem.dto.response.api

import com.bytebusters.android.mpsystem.entity.sub.UserSubscription
import com.bytebusters.android.mpsystem.entity.user.UserPreference
import java.time.LocalDateTime

class UserDto (
    var id: Long? = null,
    var email: String? = null,
    var firstName: String? = null,
    var secondName: String? = null,
    var phoneNum: String? = null,
    var isActive: Boolean? = null,
//    var createdAt: LocalDateTime? = null,
    var userPreferences: Set<UserPreference>? = null,
    var userSubscriptions: Set<UserSubscription>? = null,
    var role: RoleDto? = null
)